src/Infrastructure/Services/UserService.cs

```diff
@@ -433,61 +442,75 @@ private string CleanPhoneNumber(string phoneNumber)
+                    var htmlBody = EmailTemplate.GetHtmlTemplate().Replace("###TITLE_REPLACE1###",
+                            "Be Part of Research mobile phone number verified")
+                        .Replace("###TEXT_REPLACE1###",
+                            $"The new mobile phone number provided to secure your account has been verified. We will send a security code to this number each time you sign in.")
+                        .Replace("###TEXT_REPLACE2###",
+                            "This will not change any telephone numbers stored in the personal details section of your account. Please sign in to your account if you need to change these.")
+                        .Replace("###TEXT_REPLACE3###",
+                            "")
+                        .Replace("###TEXT_REPLACE4###",
+                            "")
+                        .Replace("###LINK_REPLACE###", "")
+                        .Replace("###LINK_DISPLAY_VALUE_REPLACE###", "block")
+                        .Replace("###TEXT_REPLACE5###",
+                            "")
+                        .Replace("###TEXT_REPLACE6###", "");
+
+                    await _emailService.SendEmailAsync(email, "Be Part of Research mobile phone number verified",
+                        htmlBody);
                 }
-            };
-            
-            // if user has a phone number send an email to confirm the change
-            if (!string.IsNullOrEmpty(existingPhoneNumber))
-            {
-                // get user email
-                var email = user.UserAttributes.FirstOrDefault(x => x.Name == "email")?.Value;
-                var htmlBody = EmailTemplate.GetHtmlTemplate().Replace("###TITLE_REPLACE1###",
-                        "Be Part of Research mobile phone number verified")
-                    .Replace("###TEXT_REPLACE1###",
-                        $"The new mobile phone number provided to secure your account has been verified. We will send a security code to this number each time you sign in.")
-                    .Replace("###TEXT_REPLACE2###",
-                        "This will not change any telephone numbers stored in the personal details section of your account. Please sign in to your account if you need to change these.")
-                    .Replace("###TEXT_REPLACE3###",
-                        "")
-                    .Replace("###TEXT_REPLACE4###",
-                        "")
-                    .Replace("###LINK_REPLACE###", "")
-                    .Replace("###LINK_DISPLAY_VALUE_REPLACE###", "block")
-                    .Replace("###TEXT_REPLACE5###",
-                        "")
-                    .Replace("###TEXT_REPLACE6###", "");
 
-                await _emailService.SendEmailAsync(email, "Be Part of Research mobile phone number verified", htmlBody);
+                await _provider.AdminUpdateUserAttributesAsync(request);
             }
+            catch (Exception e)
```

Comment:
You should catch the most specific relevant exception.
`catch(ValidationException ex)`
in which case the status code will be BadRequest rather than InternalServerError.

Might be a good idea to create a new application specific exception to throw at the end. e.g. `CognitoPhoneNumberUpdateException`. Throwing a generic `Exception` is almost always a bad idea.

Reply:
I have created the generic email message but still getting a 500 status code, perhaps you can advise?